slider/tests/test_beatmap.py

```diff
@@ -237,3 +237,66 @@ def test_hp(beatmap):
 
 def test_od(beatmap):
     assert beatmap.od() == 9
+
+
+def test_pack(beatmap):
```

Comment:
even for testing code, this is...a mess. I can barely read this code with all the line continuation characters. Either variable names need to be shortened or nesting needs to be reduced (ie code pulled out of the loop).

First though, I would say that we shouldn't be checking against `beatmap.__dict__`, but instead against each attribute explicitly. Something like...

```python
for attr in ["audio_filename", "audio_lead_in", "preview_time", ...]:
    assert gettattr(beatmap, attr) == getattr(packed_beatmap, attr)
```

with special case attributes (like hitobjects) handled separately.